### PR TITLE
Finish off builder flow for Deneb

### LINKED ISSUE #7170: Implement updated builder apis
https://github.com/ethereum/builder-specs/pull/61

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Resolved TODOs in `ExecutionBuilderModule` (basically implement getHeader and getPayload for both local, local fallback and builder + some logging)
- Enabled disabled tests + fixed current tests for deneb building
- Fixed a bug when setting the kzg commitments for blinded flow (tests are useful 🛩️  )
- Modified `ExecutionLayerChannelStub` for Deneb
- Fixed a bug where blob sidecars when unblinded simulatenously with block which caused the cached builder payload value not being available

## Fixed Issue(s)
fixes #7170 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by tbenr** [comment]
I'm testing this in my stubbed interop and I get `BuilderPayload hasn't been cached for slot N` consistently.

**Comment by StefanBratanov** [comment]
> I'm testing this in my stubbed interop and I get `BuilderPayload hasn't been cached for slot N` consistently.

Interesting. Let me give it a try as well.

File: beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockFactoryDenebTest.java [hunk 1]
```diff
@@ -117,15 +125,17 @@ void unblindSignedBlock_shouldUnblindBlockContents() {
...
     // let the unblinder return a consistent execution payload
     executionPayload =
         unblindedBeaconBlock.getMessage().getBody().getOptionalExecutionPayload().orElseThrow();
 
     final SignedBlockContainer unblindedBlockContents =
         assertBlockUnblinded(blindedBlockContents, spec);
 
-    verify(executionLayer).getCachedPayloadResult(blindedBlockContents.getSlot());
+    // make sure getCachedUnblindedPayload is second in order of method calling
+    InOrder inOrder = Mockito.inOrder(executionLayer);
```
Comment (tbenr) [inline beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockFactoryDenebTest.java]: final :D

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java [hunk 2]
```diff
@@ -391,31 +433,56 @@ private void markBuilderAsNotAvailable(final String errorMessage) {
...
   private void logReceivedBuilderExecutionPayload(final ExecutionPayload executionPayload) {
     LOG.info(
-        "Received execution payload from Builder (Block Number {}, Block Hash = {})",
+        "Received execution payload from Builder (Block Number = {}, Block Hash = {})",
         executionPayload.getBlockNumber(),
         executionPayload.getBlockHash());
   }
 
+  private void logReceivedBuilderBlobsBundle(final BlobsBundle blobsBundle) {
+    LOG.info("Received blobs bundle from Builder (Blobs = {})", blobsBundle.getNumberOfBlobs());
```
Comment (tbenr) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java]: something like Blobs count?

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java [hunk 3]
```diff
@@ -391,31 +433,56 @@ private void markBuilderAsNotAvailable(final String errorMessage) {
...
+  }
+
   private void logReceivedBuilderBid(final BuilderBid builderBid) {
     final ExecutionPayloadHeader payloadHeader = builderBid.getHeader();
+    final String blobsLog =
```
Comment (tbenr) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java:474]: here too

**Review by tbenr — APPROVED** [review]
LGTM, just nits

**Pull request merged**